File: Práctica12/PatronEstructural.py
class AntiguoSistema:
    def ejecutar(self):
        print("Ejecutando sistema antiguo")

class NuevoSistema:
    def metodo_adaptado(self):
        pass

class Adaptador(NuevoSistema):
    def __init__(self, adaptee):
        self.adaptee = adaptee
    def metodo_adaptado(self):
        self.adaptee.ejecutar()

if __name__ == "__main__":
    sistema = Adaptador(AntiguoSistema())
sistema.metodo_adaptado()

File: Práctica12/PatronEstructural.java
class AntiguoSistema {
    public void ejecutar() {
        System.out.println("Ejecutando sistema antiguo");
    }
}
interface NuevoSistema {
    void metodoAdaptado();
}
class Adaptador implements NuevoSistema {
    private AntiguoSistema adaptee;

    public Adaptador(AntiguoSistema adaptee) {
        this.adaptee = adaptee;
    }

    public void metodoAdaptado() {
        adaptee.ejecutar();
    }
}
class Main {
    public static void main(String[] args) {
        NuevoSistema sistema = new Adaptador(new AntiguoSistema());
        sistema.metodoAdaptado();
    }
}